bootloader/usb.c

```diff
@@ -437,6 +438,8 @@ static void hid_rx_callback(usbd_device *dev, uint8_t ep)
 				} while (!button.YesUp && !button.NoUp);
 			}
 			if (brand_new_firmware || button.YesUp) {
+				// check whether current firmware is signed
+				old_was_unsigned = !signatures_ok(NULL);
```

Comment:
Why backup metadata if old firmware was unsigned?  

This is not really a security issue in this case, but why keep sensitive information in memory if it is never needed.